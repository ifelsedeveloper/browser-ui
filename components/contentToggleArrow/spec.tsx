/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import ContentToggleArrow from '../contentToggleArrow'

describe('contentToggleArrow tests', () => {
  const baseComponent = (props?: object) => (
    <ContentToggleArrow id='testContentToggleArrow' summary='testSummary' {...props} />
  )

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const component = baseComponent()
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseComponent({id: 'testContentToggleArrowId'}))
      const assertion = wrapper.find('#testContentToggleArrowId').length
      expect(assertion).toBe(1)
    })
  })

  describe('component behavior', () => {
    it('can have an id', () => {
      const wrapper = shallow(baseComponent({id: 'coolId'}))
      const assertion = wrapper.props().id
      expect(assertion).toBe('coolId')
    })

    it('can have a summary', () => {
      const wrapper = shallow(baseComponent({summary: 'testSummary'}))
      const assertion = wrapper.html().includes('testSummary')
      expect(assertion).toBe(true)
    })

    it('can be opened', () => {
      const wrapper = shallow(baseComponent({open: true}))
      const assertion = wrapper.props().open
      expect(assertion).toBe(true)
    })

    it('can be closed', () => {
      const wrapper = shallow(baseComponent({open: false}))
      const assertion = wrapper.props().open
      expect(assertion).toBe(false)
    })

    it('can be set as default to open', () => {
      const wrapper = shallow(baseComponent({defaultOpen: true}))
      const assertion = wrapper.props().open
      expect(assertion).toBe(true)
    })

    it('can have a separator included', () => {
      const wrapper = shallow(baseComponent({withSeparator: true}))
      const assertion = wrapper.props().withSeparator
      expect(assertion).toBe(true)
    })

    it('can set its own state to open', () => {
      const wrapper = shallow(baseComponent())
      wrapper.setProps({open: true})
      const assertion = wrapper.state('open')
      expect(assertion).toBe(true)
    })

    it('can respond to click', () => {
      const value = {target: { open: true }}
      const onClick = jest.fn()
      const wrapper = shallow(baseComponent({onClick}))
      wrapper.find('#testContentToggleArrowControl').simulate('click', value)
      expect(onClick).toBeCalledWith(value)
    })
  })
})
