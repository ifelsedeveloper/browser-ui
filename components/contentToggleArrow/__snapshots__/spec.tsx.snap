// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contentToggleArrow tests basic tests matches the snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  width: -webkit-fill-available;
  color: inherit;
  font-size: inherit;
  font-family: inherit;
}

.c1 {
  box-sizing: border-box;
  position: relative;
  color: inherit;
  font-size: inherit;
  font-family: inherit;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1::after {
  content: "â–¶";
  font-size: 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
}

.c2 {
  box-sizing: border-box;
  color: inherit;
  font-size: inherit;
  font-family: inherit;
  margin-left: 15px;
}

.c3 {
  box-sizing: border-box;
  color: inherit;
  font-size: inherit;
  font-family: inherit;
  overflow: hidden;
  height: 0;
  width: -webkit-fill-available;
}

<div
  className="c0"
  id="testContentToggleArrow"
  open={undefined}
>
  <div
    className="c1"
    id="testContentToggleArrowControl"
    onClick={[Function]}
    open={undefined}
  >
    <div
      className="c2"
    >
      testSummary
    </div>
  </div>
  <div
    className="c3"
    open={undefined}
  />
</div>
`;
